test/fit_tests/common/fit_common.py

```diff
@@ -0,0 +1,528 @@
+    if VERBOSITY >= 9:
+        logfile_redirect = sys.stdout
+
+    (command_output, ecode) = pexpect.run(
+        cmd, withexitstatus=1,
+        events={'(?i)assword: ':ARGS_LIST['pwd'] + '\n'},
+        logfile=logfile_redirect)
+    if VERBOSITY >= 4:
+        print "scp_file_to_ora: Exit Code = {0}".format(ecode)
+
+    assert ecode == 0, \
+        'failed "{0}" because {1}. Output={2}'.format(cmd, ecode, command_output)
+    return just_fname
+
+def rackhdapi(url_cmd, action='get', payload=[], timeout=None, rest_headers={}):
+    '''
+    This routine will build URL for RackHD API, enable port, execute, and return to previous state
+    Example: fit_common.rackhdapi('/api/current/nodes') - simple 'get' command
+    Example: fit_common.rackhdapi("/api/current/nodes/ID/dhcp/whitelist", action='post')
+
+    :param url_cmd: url command for monorail api
+    :param action: rest action (get/put/post/delete)
+    :param payload: rest payload
+    :param timeout: rest timeout
+
+    :return: {'json':result_data.json(), 'text':result_data.text,
+                'status':result_data.status_code,
+                'headers':result_data.headers.get('content-type'),
+                'timeout':False}
+    '''
+
+    # unless protocol is specified, test http port, if closed, then use https
+    global API_PROTOCOL
+    global API_PORT
+    if API_PROTOCOL == None or API_PORT == None:
+        try:
+            restful("http://" + ARGS_LIST['ora'] + ":" + str(GLOBAL_CONFIG['ports']['http']) + "/")
+        except:
+            API_PROTOCOL = 'https'
+            API_PORT = GLOBAL_CONFIG['ports']['https']
+        else:
+            API_PROTOCOL = 'http'
+            API_PORT = GLOBAL_CONFIG['ports']['http']
+
+    return restful(API_PROTOCOL + "://" + ARGS_LIST['ora'] + ":" + str(API_PORT) + url_cmd,
+                       rest_action=action, rest_payload=payload, rest_timeout=timeout, rest_headers=rest_headers)
+
+
+def restful(url_command, rest_action='get', rest_payload=[], rest_timeout=None, sslverify=False, rest_headers={}):
+    '''
+    This routine executes a rest API call to the host.
+
+    :param url_command: the url that the web crawler/pusher goes
+    :param rest_action: what the restful do (get/post/put/delete)
+    :param rep_obj: report object for logging
+    :param rest_payload: payload for rest request
+    :param rest_timeout: timeout for rest request
+    :param sslverify: ssl Verify
+    :return:    {'json':result_data.json(), 'text':result_data.text,
+                'status':result_data.status_code,
+                'headers':result_data.headers.get('content-type'),
+                'timeout':False}
+    '''
+
+    result_data = None
+
+    # print URL and action
+    if VERBOSITY >= 4:
+        print "restful: Action = ", rest_action, ", URL = ", url_command
+
+    # prepare payload for XML output
+    payload_print = []
+    try:
+        json.dumps(rest_payload)
+    except ValueError:
+        payload_print = []
+    else:
```

Comment:
I get the error if use this try exception check. Seems this is not a kind of value error thus the exception could not handle it.
======================================================================
ERROR: test_api_11_update_quanta_d51_2u_skupack (test_rackhd11_api_skupack.test_rackhd11_api_skupack)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/luon1/newfit/RackHD/test/fit_tests/tests/rackhd11/test_rackhd11_api_skupack.py", line 244, in test_api_11_update_quanta_d51_2u_skupack
    self.rackhd_api_11_update_skupack("Quanta D51 2U")
  File "/home/luon1/newfit/RackHD/test/fit_tests/tests/rackhd11/test_rackhd11_api_skupack.py", line 97, in rackhd_api_11_update_skupack
    api_response = fit_common.rackhdapi('/api/1.1/skus/'+ item['id']+'/pack', action='binary-put',payload=f)
  File "/home/luon1/newfit/RackHD/test/fit_tests/common/fit_common.py", line 254, in rackhdapi
    rest_action=action, rest_payload=payload, rest_timeout=timeout, rest_headers=rest_headers)
  File "/home/luon1/newfit/RackHD/test/fit_tests/common/fit_common.py", line 282, in restful
    json.dumps(rest_payload)
  File "/usr/lib/python2.7/json/__init__.py", line 243, in dumps
    return _default_encoder.encode(obj)
  File "/usr/lib/python2.7/json/encoder.py", line 207, in encode
    chunks = self.iterencode(o, _one_shot=True)
  File "/usr/lib/python2.7/json/encoder.py", line 270, in iterencode
    return _iterencode(o, 0)
  File "/usr/lib/python2.7/json/encoder.py", line 184, in default
    raise TypeError(repr(o) + " is not JSON serializable")
TypeError: <open file u'quanta-d51-2u_d51.tar.gz', mode 'rb' at 0x7fdcc77e0a50> is not JSON serializable